onnxruntime/core/providers/dnnl/subgraph/dnnl_gemm.cc

```diff
@@ -92,14 +93,17 @@ void DnnlGemm::CreatePrimitive(DnnlSubgraphPrimitive& sp, DnnlNode& node) {
   dnnl::primitive_attr matmul_attr;
   // scale the output from MatMul to alpha
   float alpha = GetAlpha(node);
-  std::vector<float> alphaScale({alpha});
-  matmul_attr.set_output_scales(0, alphaScale);
+  // Set the value to scale DNNL_ARG_SRC with mask 0
+  matmul_attr.set_scales_mask(DNNL_ARG_SRC, 0);
+  // Create the memory object related to the scale
+  auto alpha_mem = dnnl::memory({{1}, dnnl::memory::data_type::f32, {1}}, eng);
+  // Write the alpha value into the memory object 
+  dnnl_util::WriteToDnnlMemory<float>(alpha_mem, {alpha});
 
   auto matmul_dst_md = dnnl::memory::desc(output_shape, node.Output(OUT_Y).Type(), {N, 1});
 
-  auto matmul_d = dnnl::matmul::desc(a_md, b_md, matmul_dst_md);
   dnnl::matmul::primitive_desc matmul_pd;
-  matmul_pd = dnnl::matmul::primitive_desc(matmul_d, matmul_attr, eng);
+  matmul_pd = dnnl::matmul::primitive_desc(eng, a_md, b_md, matmul_dst_md, matmul_attr);
```

Comment:
Just checking, did oneDNN 3.0 added a attribute to the matmul. There is not default matmul that can be called without the new attribute and we must set the scales mask attribute since we are using the alpha attribute correct?

Reply:
Yes, another thing to note is that some primitives don't support scaling the output (as we did before with _matmul_attr.set_output_scales(0, alphaScale);_ ) so in this case we scaled one input, this has the same result because of the operation